src/safeposix/syscalls/fs_calls.rs

```diff
@@ -26,33 +51,48 @@ impl Cage {
     }
 
     pub fn open_syscall(&self, path: &str, flags: i32, mode: u32) -> i32 {
-        //Check that path is not empty
+        // Check that the given input path is not empty
         if path.len() == 0 {
             return syscall_error(Errno::ENOENT, "open", "given path was null");
         }
+        
+        // Retrieve the absolute path from the root directory. The absolute path is then used to validate directory paths
+        // while navigating through subdirectories and creating a new file or open existing file at the given location.
         let truepath = normpath(convpath(path), self);
 
+        // Fetch the next file descriptor and its lock write guard to ensure the file can be associated with the file descriptor 
         let (fd, guardopt) = self.get_next_fd(None);
+        // Return an error when no file descriptor is available
         if fd < 0 {
-            return fd;
+            return syscall_error(
+                Errno::ENFILE,
+                "open_helper",
+                "no available file descriptor number could be found",
+            );
         }
+        // File Descriptor Write Lock Guard
         let fdoption = &mut *guardopt.unwrap();
 
+        // Walk through the absolute path which returns a tuple consisting of inode number of file (if it exists), and inode number of parent (if it exists)
         match metawalkandparent(truepath.as_path()) {
-            //If neither the file nor parent exists
+            // Case 1: When neither the file directory nor the parent directory exists
```

Comment:
It's sort of weird to structure it like this because if the parent director doesn't exist, then the file clearly won't.  Can you just check to make sure the parent dir exists and return an error otherwise?

Reply:
If you just want clarity the tuple check in match statement would just be (.., None) since the second field is the parent and the first field doesn't matter. Thought metawalkparent will always return (None, None) if the parent doesn't exist since like you said if we don't find a parent we obviously don't have the file.